### ISSUE TITLE
imfilter_fft with 

### ISSUE THREAD

**Issue by andferrari** [timholy/Images.jl#399]
Thank you for this amazing package !

 It seems to me that when `img` and `kern` have the same size, the circular filtering should be simply `ifft(fft(img).*fft(kern))`. But it seems that `imfilter_fft`does a circular padding of `img` and a zero padding of `kern`. 

The final result is the same but the time and memory not!
```julia
julia> img = randn(2048,2048); kern = img;

julia> @time imfilter_fft(img, kern, "circular");
 12.752258 seconds (342 allocations: 1.813 GB, 1.25% gc time)

julia> @time ifft(fft(img).*fft(kern));
  2.591529 seconds (241 allocations: 384.011 MB)
```

**Comment by timholy** [comment]
Definitely would be an improvement! That function was written not expecting the kernel to be so large, but of course you're right that this is an obvious and important case. Good catch.

Care to submit a fix? If not, I or someone will implement it when time allows.

**Comment by andferrari** [comment]
I'll give a try!

**Comment by timholy** [comment]
Awesome! Let me know if you run into trouble. If you're not experienced with `git`, that can be a barrier; if this describes you, [these docs](http://docs.julialang.org/en/release-0.4/manual/packages/#package-development) may help. The short version is
```
Pkg.checkout("Images")
cd(Pkg.dir("Images"))
;git checkout -b myfixes
<edit code>
;git commit -a -m "Make imfilter_fft a lot better"
Pkg.submit("Images")
```

**Comment by timholy** [comment]
Closed by #400
